test/extended/router/h2spec.go

```diff
@@ -0,0 +1,289 @@
+	defer os.Remove(outputFile.Name())
+
+	g.By("Running the h2spec CLI test")
+	h2specOutput, h2specErr := e2e.RunHostCmd(oc.KubeFramework().Namespace.Name, podName, t.h2specCommand(outputFile.Name()))
+	if h2specErr != nil {
+		// We don't assert on a failure here; h2spec will exit
+		// with non-zero if _any_ test in the suite fails. We
+		// later assert on failing to copy the results,
+		// failing to read the results, failing to decode the
+		// results because there are no testsuites in the
+		// decoded results. We always expect at least one
+		// testsuite in the results.
+		e2e.Logf("[%s] h2spec: error=\n%s", t.routeType, h2specErr)
+	}
+	if h2specErr != nil && h2specErr.Error() != h2specOutput {
+		// Also log any output if different.
+		e2e.Logf("[%s] h2spec: output=\n%s", t.routeType, h2specOutput)
+	}
+
+	g.By(fmt.Sprintf("[%s] Fetching %q from the container", t.routeType, outputFile.Name()))
+	rsyncErr := oc.Run("rsync").Args(podName+":"+outputFile.Name(), filepath.Dir(outputFile.Name()), "--strategy=tar").Execute()
+	o.Expect(rsyncErr).NotTo(o.HaveOccurred())
+
+	g.By(fmt.Sprintf("[%s] Reading %q", t.routeType, outputFile.Name()))
+	data, readErr := ioutil.ReadFile(outputFile.Name())
+	o.Expect(readErr).NotTo(o.HaveOccurred())
+
+	g.By(fmt.Sprintf("[%s] Decoding %q", t.routeType, outputFile.Name()))
+	testSuites, err := h2spec.DecodeJUnitReport(strings.NewReader(string(data)))
+	o.Expect(err).NotTo(o.HaveOccurred())
+	o.Expect(len(testSuites)).ShouldNot(o.BeZero(), "expected > 0 testSuites")
+
+	return testSuites
+}
+
+func runConformanceTestsAndLogAggregateFailures(oc *exutil.CLI, t h2specRouteTypeTest, iterations int) {
+	sortKeys := func(m map[string]int) []string {
+		var index []string
+		for k := range m {
+			index = append(index, k)
+		}
+		sort.Strings(index)
+		return index
+	}
+
+	printFailures := func(prefix string, m map[string]int) {
+		for _, id := range sortKeys(m) {
+			e2e.Logf("[%s] %sTestCase ID: %q, cumulative failures: %v", t.routeType, prefix, id, m[id])
+		}
+	}
+
+	failuresByTestCaseID := map[string]int{}
+
+	for i := 1; i <= iterations; i++ {
+		failures := failingTests(runConformanceTests(oc, t))
+		e2e.Logf("[%s] Iteration %v/%v: had %v failures", t.routeType, i, iterations, len(failures))
+
+		// Aggregate any new failures
+		for _, f := range failures {
+			failuresByTestCaseID[f.ID()]++
+		}
+
+		// Dump the current state at every iteration should
+		// you wish to interrupt/abort the running test.
+		printFailures("\t", failuresByTestCaseID)
+	}
+
+	e2e.Logf("[%s] Sampling completed: %v test cases failed", t.routeType, len(failuresByTestCaseID))
+	printFailures("", failuresByTestCaseID)
+}
+
+func getHostnameForRoute(oc *exutil.CLI, routeName string) string {
+	var hostname string
+	ns := oc.KubeFramework().Namespace.Name
+	err := wait.Poll(time.Second, changeTimeoutSeconds*time.Second, func() (bool, error) {
+		e2e.Logf("Getting hostname for route %q", routeName)
```

Comment:
This could be spammy... consider only logging on failure?

Reply:
Addressed in https://github.com/openshift/origin/pull/24674/commits/2ecf50fbc41867d1be158f77fb4167d15c083b2f